app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -35,6 +37,42 @@ def events
       respond_for_index :collection => collection
     end
 
+    api :POST, "/hosts/subscriptions/register", N_("Simulate a host subscription-manager registration")
```

Comment:
If you expect users to use the rhsm register api endpoint then you have to realize that:

a) it is not documented
b) We don't control what the api accepts.  Candlepin could add a parameter and it wouldn't matter to that api one bit.  That is part of the purpose of the /rhsm/ controller. 
b) It uses terminology that is not native to katello.  Environment doesn't mean Lifecycle Environment or Puppet Environment. There is lots of camel casing and not snake casing.  Its entirely inconsistent with our other apis.

I seriously disagree that that is an appropriate API for a user to use. 

As for your argument is that there are no use cases, see @thomasmckay not so detailed list below :)  
 3) splice, 4) spacewalk to katello migration
are two that are used with current (and older) katello versions.  

The amount of duplicated code is tiny (two lines). 

You could argue that this code is temporary until we can easily add and remove facets from hosts, but there is functionality that 'registering' takes care of such as making sure that a pulp consumer is created with the same UUID as the candlepin consumer.  This would have to be done manually by the user if they were 'adding' one facet at a time and would be nearly impossible if each facet is treated independently and created host creation time.   At some point in the future maybe these two things don't need to have the same UUID (the pulp UUID may become unimportant) but we are not there today.  The candlepin environment also needs to match the content facet's content view and lifecycle environment.  These are two things that 'bleed' between the facets.